kratos/processes/calculate_distance_to_skin_process.cpp

```diff
@@ -236,8 +234,20 @@ namespace Kratos
             }
 
             distances[i_direction] *= ray_color;
+			if (ray_color == -1) {
+				n_ray_neg++;
+			} else {
+				n_ray_pos++;
+			}
         }
 
+		// Check the obtained cartesian ray colors
+		// If this situation happens, do the "evolved Predator" raycasting to vote
+		if (n_ray_neg != 0 && n_ray_pos != 0) {
+			const double ray_epsilon = 1e-4;
```

Comment:
@roigcarlo  @pooyan-dadvand @RiccardoRossi Guys, what about this epsilon? Is the one used to perturb the origin ray. Maybe it's a good idea to create an extra constructor with it and if it is not provided it takes a default value.

Reply:
Is it in normalize space (key) or in real space (coordinate) dimension? 

In my opinion this epsilon should be very small. Much smaller than this.... But a more than the epsilon of intersection test.

Reply:
in my opinion the epsilon should be either very small or relative to a given measure of the size...

one option is to expose it as a parameter to the user

Reply:
I would go for one default value and expose to change by user...

Reply:
Guys, I've just implemented a new constructor that allows the user to pass the extra rays epsilon.